pkg/actions/actions.go

```diff
@@ -0,0 +1,291 @@
+	Zip("")
+
+	logrus.Info("Doing a final check to ensure Git is clean")
+	checkGit()
+
+	logrus.Info("Successfully validated current repository!")
+}
+
+func ValidateLocal() {
+	logrus.Infof("Checking if Git is clean")
+	checkGit()
+
+	logrus.Infof("Generating charts")
+	Charts("")
+
+	logrus.Infof("Checking if Git is clean after generating charts")
+	checkGit()
+	logrus.Infof("Successfully validated that current charts and assets are up to date.")
+
+	logrus.Infof("Running local validation only, skipping pulling upstream")
+
+	logrus.Info("Zipping charts to ensure that contents of assets, charts, and index.yaml are in sync.")
+	Zip("")
+
+	logrus.Info("Doing a final check to ensure Git is clean")
+	checkGit()
+
+	logrus.Info("Successfully validated current repository!")
+}
+
+func ValidateRemote() {
+	chartsScriptOptions := parseScriptOptions()
+
+	logrus.Infof("Checking if Git is clean")
+	checkGit()
+
+	logrus.Infof("Running remote validation only, skipping generating charts locally")
+
+	pullUpstream(chartsScriptOptions)
+
+	logrus.Info("Zipping charts to ensure that contents of assets, charts, and index.yaml are in sync.")
+	Zip("")
+
+	logrus.Info("Doing a final check to ensure Git is clean")
+	checkGit()
+
+	logrus.Info("Successfully validated current repository!")
+}
+
+func Standardize() {
+	repoRoot := getRepoRoot()
+	repoFs := filesystem.GetFilesystem(repoRoot)
+	if err := standardize.RestructureChartsAndAssets(repoFs); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func Template() {
+	repoRoot := getRepoRoot()
+	repoFs := filesystem.GetFilesystem(repoRoot)
+	chartsScriptOptions := parseScriptOptions()
+	if err := update.ApplyUpstreamTemplate(repoFs, *chartsScriptOptions); err != nil {
+		logrus.Fatalf("Failed to update repository based on upstream template: %s", err)
+	}
+	logrus.Infof("Successfully updated repository based on upstream template.")
+}
+
+func getRepoRoot() string {
+	repoRoot, err := os.Getwd()
+	if err != nil {
+		logrus.Fatalf("Unable to get current working directory: %s", err)
+	}
+	return repoRoot
+}
+
+func getPackages(currentPackage string) []*charts.Package {
+	repoRoot := getRepoRoot()
+	packages, err := charts.GetPackages(repoRoot, currentPackage)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	return packages
+}
+
+func parseScriptOptions() *options.ChartsScriptOptions {
+	configYaml, err := ioutil.ReadFile(ChartsScriptOptionsFile)
+	if err != nil {
+		logrus.Fatalf("Unable to find configuration file: %s", err)
+	}
+	chartsScriptOptions := options.ChartsScriptOptions{}
+	if err := yaml.UnmarshalStrict(configYaml, &chartsScriptOptions); err != nil {
+		logrus.Fatalf("Unable to unmarshall configuration file: %s", err)
+	}
+	return &chartsScriptOptions
+}
+
+func checkGit() {
+	_, _, status := getGitInfo()
+	if !status.IsClean() {
+		logrus.Warnf("Git is not clean:\n%s", status)
+		logrus.Fatal("Repository must be clean to run validation")
+	}
+}
```

Comment:
I think this should be modified to `checkIfGitClean` or something more indicative of what we are checking Git for. I also think we should log the `Git is clean` log here itself.